arch/arm/core/cortex_m/reset.S

```diff
@@ -61,12 +63,44 @@ SECTION_SUBSEC_FUNC(TEXT,_reset_section,__start)
 #endif
 
     /* lock interrupts: will get unlocked when switch to main task */
-    bl	lock_interrupts
+#if defined(CONFIG_ARMV6_M_ARMV8_M_BASELINE)
+    cpsid i
+#elif defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE)
+    movs.n r0, #_EXC_IRQ_DEFAULT_PRIO
+    msr BASEPRI, r0
+#else
+#error Unknown ARM architecture
+#endif
+
 #ifdef CONFIG_WDOG_INIT
     /* board-specific watchdog initialization is necessary */
     bl _WdogInit
 #endif
 
+#ifdef CONFIG_INIT_STACKS
+    ldr r0, =_interrupt_stack
+    ldr r1, =0xaa
+    ldr r2, =CONFIG_ISR_STACK_SIZE
```

Comment:
One issue to consider, @andrewboie , un-related to the patch: we might be losing bytes, here, because _interrupt_stack is defined using K_THREAD_STACK_DEFINE, which adds, possibly a guard area. But this is not taken into account in CONFIG_ISR_STACK_SIZE - that's the original ISR stack configuration.

I propose to _not_ define the interrupt stack with K_THREAD_STACK_DEFINE, or add some smart options in the macro, to not include the guard area. Otherwise, we should use K_THREAD_STACK_SIZEOF(_interrupt_stack), but that we can do if we keep the function in C code.

Reply:
Didn't the old code do this?
I don't think we should refector this in this PR

Reply:
The old code did not do this, we always had this problem, unfortunately.
Sure, we should not fix this in the PR, I stressed that this is un-related, but, actually, we need to discuss _how_ to fix it (and with respect, also, to how and if we refactor the STACK_DEFINE macros). I'll post this to an existing/new issue.

Reply:
Resolving this - I am tracking down the issue with the ISR size in #16210